File: 5차 과제/6-1/C 소스 파일/6-1.c
#include <stdio.h>
#include "6-1.h"

elment1 front=-1;
elment1 rear=-1;
elment2 queue[size];

int main() {
    printf("*****순차 큐 연산*****\n");
    enqueue(queue, 'A');
    enqueue(queue, 'B');
    enqueue(queue, 'C');
    printf("peek item: %c\n\n", queue[front + 1]);

    elment2 item = dequeue(queue);
    printf("\t삭제 데이터: %c\n", item);
    item = dequeue(queue);
    printf("\t삭제 데이터: %c\n", item);
    item = dequeue(queue);
    printf("\t\t삭제 데이터: %c\n", item);
    enqueue(queue, 'D');
    enqueue(queue, 'E');
}

File: DataStructure/5차 과제/6-1/py 소스 파일/6_1.py
def main():
    queue=[]
    queue.append('A')
    print(f"삽입A>> queue:{queue}")
    queue.append('B')
    print(f"삽입B>> queue:{queue}")
    queue.append('C')
    print(f"삽입C>> queue:{queue} peek item:{queue[0]}\n")

    item=queue.pop(0)
    print(f"삭제>> queue:{queue}\t삭제 데이터:{item}")
    item=queue.pop(0)
    print(f"삭제>> queue:{queue}\t삭제 데이터:{item}")
    item=queue.pop(0)
    print(f"삭제>> queue:{queue}\t\t삭제 데이터:{item}")

    queue.append('D')
    print(f"삽입D>> queue:{queue}")
    print(f"삽입E>> queue is full\n\tqueue:{queue}")
if __name__ == "__main__":
    main()